app/controllers/labels_controller.rb

```diff
@@ -2,13 +2,43 @@
 
 class LabelsController < ApplicationController
   load_and_authorize_resource
+
+  rescue_from CanCan::AccessDenied do |_exception|
+    redirect_to root_path, alert: t('controllers.authorization')
+  end
+
+  def index; end
+
+  def update
+    @label.update(params.permit(:color))
+  end
+
+  def destroy
+    @label.destroy
+  end
```

Comment:
Should we use update! and destroy! instead, catch the error and return the error message?